### ISSUE TITLE
property extensions lose type information

### ISSUE THREAD

**Issue by jackd** [microsoft/pyright#307]
Methods annotated with `@property` preserve type information, but this is not the case for methods annotated by classes derived from `property` (e.g. `abc.abstractproperty`).

The following example obviously has issues in terms of instantiating an `ABC`, but I've chosen it because it's the simplest case of a class which extends `property` (my actual motivational case is for a `memoized_property`).

```python
import abc

def f(x: int):
    pass


class Base(object):
    @property
    def x(self) -> int:
        return 3

    @property
    def y(self) -> float:
        return 3.5


class Abstract(abc.ABC):
    @abc.abstractproperty
    def x(self) -> int:
        raise NotImplementedError

    @abc.abstractproperty
    def y(self) -> float:
        raise NotImplementedError


b = Base()
f(b.x)
f(b.y)  # <- complains as expected

a = Abstract()
f(a.x)
f(a.y)  # <- no complaint.
```

(otherwise loving working with `pyright` - thanks!)

**Comment by erictraut** [comment]
Thanks for the bug report. I wasn't aware of the @abc.abstractproperty before you mentioned it. I've added the necessary support, and it will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.0.81, which I just published

**Comment by jackd** [comment]
@erictraut thanks for the fix, but my actual issue is with general property extensions, rather than just `abstractproperty` (I was using `abstractproperty` as a minimal example). For example, the following still fails to complain:

```python
def f(x: int):
    pass


class custom_property(property):
    pass


class Custom(object):

    @custom_property
    def x(self) -> int:
        return 3

    @custom_property
    def y(self) -> float:
        return 3.5


m = Custom()
f(m.x)  #  <- no complaint as expected
f(m.y)  #  <- no complaint, but should - y is float, f expects int
```

My actual use case at the moment is `memoized_property`
```python
class memoized_property(property):  # pylint: disable=invalid-name
    """Descriptor that mimics @property but caches output in member variable."""

    def __get__(self, obj: Any, type: Optional[type] = None) -> Any:
        # See https://docs.python.org/3/howto/descriptor.html#properties
        if obj is None:
            return self
        if self.fget is None:
            raise AttributeError("unreadable attribute")
        attr = "__cached_" + self.fget.__name__
        cached = getattr(obj, attr, None)
        if cached is None:
            # cached = self.fget(obj)
            cached = super(memoized_property, self).__get__(obj, type)
            setattr(obj, attr, cached)
        return cached
```

**Comment by erictraut** [comment]
Thanks again for reporting this problem. It will be fixed in the next version of pyright.

**Comment by erictraut** [comment]
This is now fixed in 1.1.4, which I just published.
